### ISSUE TITLE
Stored XSS at function Add discussion in OJS <= 3.4

### ISSUE THREAD

**Issue by g0dd0ghd** [pkp/pkp-lib#9599]
**Describe the bug**
I tested the demo site you provided and I see that there is a stored XSS in the function Add discussion. Although it was fixed for the previous bug https://github.com/pkp/pkp-lib/commit/a0485a5c7410821e0432c079ca676ee6026bffb9 by adding function `htmlspecialchars`, the `<` and `>` in `$headNote->getTitle()` content still not be converted to HTML entities because you need to reassign a value of `$headNote->getTitle()`.

<img width="456" alt="image" src="https://github.com/pkp/pkp-lib/assets/75677317/092e2042-0752-452d-b123-0e74600079b9">

**Proof of Concept**
```JavaScript
dang"><script>alert(document.location)</script>
````
**To Reproduce**
1. Login as any user
2. Go to tab Submissions
3. Click View any submission
4. In any tab, scroll down and click Add discussion 
5. Insert payload to Subject field
6. After creating, when other users click on the discussion, the payload will be executed

**Impact**
This security vulnerability has the potential to steal multiple users' cookies, gain unauthorized access to that user's account through stolen cookies, or redirect the user to other malicious websites...

**PoC video**
https://studenthcmusedu-my.sharepoint.com/:v:/g/personal/20120264_student_hcmus_edu_vn/ESx-S_CLzhBPmxGBnPE1cM8BQKjr_UwNJjtwHTNMyGw_Ig?nav=eyJyZWZlcnJhbEluZm8iOnsicmVmZXJyYWxBcHAiOiJPbmVEcml2ZUZvckJ1c2luZXNzIiwicmVmZXJyYWxBcHBQbGF0Zm9ybSI6IldlYiIsInJlZmVycmFsTW9kZSI6InZpZXciLCJyZWZlcnJhbFZpZXciOiJNeUZpbGVzTGlua0NvcHkifX0&e=bgaTOm

**Solution**
1. Reassigning `$headNote->getTitle() = htmlspecialchars($headNote->getTitle())`
2. Filter input data.
3. Encode data on output.
4. Content Security Policy (CSP)

**Comment by asmecher** [comment]
@g0dd0ghd, I believe this has already been fixed -- testing with a fresh checkout from the `stable-3_4_0` branch does not result in an XSS leak being displayed. What version are you testing with?

Please do not report potential security issues via Github issues; see [`docs/SECURITY`](https://github.com/pkp/ojs/blob/main/SECURITY.md) for responsible reporting guidelines.

**Comment by g0dd0ghd** [comment]
@asmecher, Sorry for my confusion in the initial report,  the demo site I tested was running version 3.3.0 and I also misunderstood the code. BUT when I rechecked the demo website running version 3.4.0.3 (this [site](https://ojs34.testdrive.publicknowledgeproject.org/index.php/testdrive-journal)) provided on pkp [page](https://pkp.sfu.ca/software/ojs/demo/), the "Add discussion" function was still vulnerable to XSS with the same payload above. Here is my screenshot about it.

<img width="895" alt="image" src="https://github.com/pkp/pkp-lib/assets/75677317/80dfcffa-1ec7-41b4-bcae-f11026401062">

I don't know if the demo site is updated to the latest version or not. 
Thanks for your response.
